development/extensions/tutorial.rst

```diff
@@ -0,0 +1,372 @@
+        },
+        "require-dev": {
+            "phpbb/epv": "dev-master"
+        },
+        "extra": {
+            "display-name": "Acme Demo Extension",
+            "soft-require": {
+                "phpbb/phpbb": "~3.1"
+            }
+        }
+    }
+
+.. csv-table::
+   :header: "Field", "Content"
+   :delim: |
+
+   ``name`` | "Must contain the vendor and extension name, matching the folder
+   structure"
+   ``type`` | ``phpbb-extension`` **must not be changed**
+   ``description`` | "A short description of your extension, may also be empty
+   (but not skipped)"
+   ``homepage`` | "Must contain a valid URL. It is recommended to use the link
+   to the contribution in the customisation database, or to the repository of
+   your extension (if you are using a public one like GitHub)."
+   ``version`` | "The version of your extension. You may also use ``-a1``
+   (alpha), ``-b1`` (beta), ``-rc1`` (release candidate) and ``-dev`` suffixes
+   (although you may not submit them to the customisation database.)"
+   ``time`` | "Must contain the release date of this version (date is required,
+   time is optional)"
+   ``license`` | The license must be ``GPL-2.0`` for now
+   ``authors`` | "An array with the authors of the extension.
+   See `authors`_ for more details."
+   ``require`` | "An array with requirements of the extension.
+   See `require`_ for more details."
+   ``require-dev`` | "An array with development requirements of the extension.
+   See `require-dev`_ for more details."
+   ``extra`` | "An array with additional values of the extension.
+   See `extra`_ for more details."
+
+authors
+-------
+
+You may have unlimited authors. But you should at least have one.
+
+.. csv-table::
+   :header: "Field", "Content"
+   :delim: |
+
+   ``name`` | Name of the author
+   ``email`` | Email address of the author
+   ``homepage`` | Must contain one valid URL or be empty
+   ``role`` | "Role can be used to specify what the authors did for the
+   extension (e.g. Developer, Translator, Supporter, ...)"
+
+require
+-------
+
+In the ``require`` section you can specify the technical requirements of your
+extension. Examples are the ``php`` version, or
+`third party libraries <https://packagist.org/>`_. Since our demo extension does
+not require any additional library, we only use the PHP version requirement, to
+make sure people have the right PHP version on their server, and composer
+installers for some internal handling. phpBB 3.1 requires PHP 5.3.3 or higher,
+so the version comparison is ``>= 5.3.3``.
+
+require-dev
+-----------
+
+In the ``require-dev`` section you can specify the technical requirements of your
+extension, which are only required for development. We use the
+`Extension PreValidation <https://packagist.org/packages/phpbb/epv>`_ Tool from
+the phpBB Extension team here, to perform some basic validation when running our
+tests on Travis CI later. Since we always want to have the newest version, we
```

Comment:
Maybe you can clarify what Travis CI is by saying something like "running our tests automatically" and maybe something like "on the hosted Continuous Integration service: Travis CI"